File: VisualStudio/CSharp/w7cpframe/clsPfConnectivity.cs
//
// Copyright (c)2016 S C Harrison
// Refer to License.txt for terms and conditions of use.
//
using System;
using System.Collections.Generic;
using System.Text;
using System.IO;
using System.Text.RegularExpressions;

namespace w7cpframe
{
    class clsPfConnectivity
    {
        //Define Class

        public int key = 0;

        public int jj = 0;            //.. joint No. @ end "j" of a member ..  [na]
        public int jk = 0;            //.. joint No. @ end "k" of a member ..  [nb]
        public int sect = 0;          //.. section group of member ..          [ns]
        public int rel_i = 0;         //.. end i release of member ..          [mra]
        public int rel_j = 0;         //.. end j release of member ..          [mrb]

        public double L = 0;             //Length of Member

        public clsPfForce jnt_jj; //= null; //new clsPfForce; // clsPfForce
        public clsPfForce jnt_jk; //= null; //new clsPfForce; // clsPfForce



        public void initialise()
        {
            key = 0;
            jj = 0;
            jk = 0;
            sect = 0;
            rel_i = 0;
            rel_j = 0;

            //  this.jnt_jj = new clsPfForce;
            //  this.jnt_jj.initialise;
            //  
            //  this.jnt_jk = new clsPfForce;
            //  this.jnt_jk.initialise;

        }

        public void setValues(int memberKey, int NodeA, int NodeB, int sectionKey, int ReleaseA, int ReleaseB)
        {
            key = memberKey;
            jj = NodeA;
            jk = NodeB;
            sect = sectionKey;
            rel_i = ReleaseA;
            rel_j = ReleaseB;
        }

        public string sprint()
        {
            string s;

            s = "";
            s = s + key.ToString().PadLeft(8);
            s = s + jj.ToString().PadLeft(6);
            s = s + jk.ToString().PadLeft(6);
            s = s + sect.ToString().PadLeft(6);
            s = s + rel_i.ToString().PadLeft(6);
            s = s + rel_j.ToString().PadLeft(2);

            return s;

        }

        public void cprint()
        {
            System.Console.WriteLine(sprint());
            //  jnt_jj.cprint
            //  jnt_jk.cprint

        }

        public void fprint(StreamWriter fp)
        {
            fp.WriteLine(sprint());
        }

        public void fgetData(StreamReader fp)
        {
            string s;

            System.Console.WriteLine("fgetData ...");

            s = fp.ReadLine();
            //System.Console.WriteLine( s);
            sgetData(s);

            System.Console.WriteLine("... fgetData");
        }

        public void sgetData(string s)
        {
            string[] dataflds = new string[10]; //(0 To 9);
            char[] delimters = new char[] { ' ', '\t' };
            int i;

            System.Console.WriteLine("sgetData ...");

            System.Console.WriteLine(s);

            string regResult;
            string pattern = @"^\s+|\s+$"; //trim trailing spaces
            regResult = Regex.Replace(s, pattern, "");

            dataflds.Initialize();
            string pattern2 = @"-?\d+(?:[,.]\d+)?";
            i = 0;
            foreach (Match match in Regex.Matches(regResult, pattern2))
            {
                if (match.Value != "")
                {
                    dataflds[i] = match.Value;
                    System.Console.WriteLine(match.Value);
                    i++;
                }
            }


            //   n = dataflds.length;
            //   for (i=0;i<n;i++)
            //   {
            //    System.Console.WriteLine( i + "<" + dataflds[i] + ">" );
            //   }
            //   
            key = int.Parse(dataflds[0]);
            jj = int.Parse(dataflds[1]);
            jk = int.Parse(dataflds[2]);
            sect = int.Parse(dataflds[3]);
            rel_i = int.Parse(dataflds[4]);
            rel_j = int.Parse(dataflds[5]);

            System.Console.WriteLine("... sgetData");
        }



    } //class
} //name space

File: JScript/clsPfConnectivity.js
//
// Copyright (c)2014 S C Harrison
// Refer to License.txt for terms and conditions of use.
//


//Define Class
function clsPfConnectivity()
{
  key=0;
  
   jj = 0;            //.. joint No. @ end "j" of a member ..  [na]
   jk = 0;            //.. joint No. @ end "k" of a member ..  [nb]
   sect = 0;          //.. section group of member ..          [ns]
   rel_i = 0;         //.. end i release of member ..          [mra]
   rel_j = 0;         //.. end j release of member ..          [mrb]
   
   L = 0;             //Length of Member
   
   jnt_jj = null; //new clsPfForce; // clsPfForce
   jnt_jk = null; //new clsPfForce; // clsPfForce
}


clsPfConnectivity.prototype.initialise = function()
{
  this.key=0;
  this.jj = 0;
  this.jk = 0;
  this.sect = 0;
  this.rel_i = 0;
  this.rel_j = 0;
  
//  this.jnt_jj = new clsPfForce;
//  this.jnt_jj.initialise;
//  
//  this.jnt_jk = new clsPfForce;
//  this.jnt_jk.initialise;

}

clsPfConnectivity.prototype.setValues = function(memberKey, NodeA, NodeB, sectionKey, ReleaseA, ReleaseB)
{
  this.key = memberKey;
  this.jj = NodeA;
  this.jk = NodeB;
  this.sect = sectionKey;
  this.rel_i = ReleaseA;
  this.rel_j = ReleaseB;
}

clsPfConnectivity.prototype.sprint = function()
{
  var s;
  
  s = "";
  s = s + StrLPad(this.key.toString(), 8)
  s = s + StrLPad(this.jj.toString(), 6)
  s = s + StrLPad(this.jk.toString(), 6)
  s = s + StrLPad(this.sect.toString(), 6)
  s = s + StrLPad(this.rel_i.toString(), 6)
  s = s + StrLPad(this.rel_j.toString(), 2)

  return s

}

clsPfConnectivity.prototype.cprint = function()
{
  WScript.Echo(this.sprint());
//  jnt_jj.cprint
//  jnt_jk.cprint

}

clsPfConnectivity.prototype.fprint = function(fp)
{
  fp.WriteLine(this.sprint());
}

clsPfConnectivity.prototype.fgetData = function(fp)
{
  var s;
  var n;
  var dataflds;
  var i,n;

  WScript.Echo( "fgetData ...");
  
  s = fp.ReadLine();
  //WScript.Echo( s);
  this.sgetData(s);
   
  WScript.Echo( "... fgetData");
}

clsPfConnectivity.prototype.sgetData = function(s)
{
  var n;
  var dataflds;
  var i,n;

  WScript.Echo( "sgetData ...");

  WScript.Echo( s);

   s = s.replace(/^\s+|\s+$/gm,''); //trim trailing spaces
   dataflds = s.match(/\s*(\d*)\s*/gi);
   
   
//   n = dataflds.length;
//   for (i=0;i<n;i++)
//   {
//    WScript.Echo( i + "<" + dataflds[i] + ">" );
//   }
//   
    this.key = parseInt(dataflds[0]);
    this.jj = parseInt(dataflds[1]);
    this.jk = parseInt(dataflds[2]);
    this.sect = parseInt(dataflds[3]);
    this.rel_i = parseInt(dataflds[4]);
    this.rel_j = parseInt(dataflds[5]);
   
   WScript.Echo( "... sgetData");
}
